testerum-backend/testerum-runner/testerum-runner-junit/src/main/kotlin/com/testerum/runner/junit5/provider/JunitTestProvider.kt

```diff
@@ -0,0 +1,198 @@
+package com.testerum.runner.junit5.provider
+
+import com.testerum.common_jdk.stopwatch.StopWatch
+import com.testerum.model.tests_finder.TestPath
+import com.testerum.runner.events.execution_listener.ExecutionListener
+import com.testerum.runner.events.model.RunnerEvent
+import com.testerum.runner.events.model.TestEndEvent
+import com.testerum.runner.events.model.TestStartEvent
+import com.testerum.runner.events.model.TextLogEvent
+import com.testerum.runner.events.model.log_level.LogLevel
+import com.testerum.runner.events.model.position.EventKey
+import com.testerum.runner_cmdline.cmdline.params.model.CmdlineParams
+import com.testerum.runner_cmdline.events.execution_listeners.junit.JUnitExecutionListener
+import com.testerum.runner_cmdline.events.execution_listeners.utils.console_output_capture.ConsoleOutputCapturer
+import com.testerum.runner_cmdline.module_di.RunnerModuleBootstrapper
+import com.testerum.runner_cmdline.runner_tree.nodes.RunnerFeatureOrTest
+import com.testerum.runner_cmdline.runner_tree.nodes.feature.RunnerFeature
+import com.testerum.runner_cmdline.runner_tree.nodes.parametrized_test.RunnerParametrizedTest
+import com.testerum.runner_cmdline.runner_tree.nodes.parametrized_test.RunnerScenario
+import com.testerum.runner_cmdline.runner_tree.nodes.suite.RunnerSuite
+import com.testerum.runner_cmdline.runner_tree.nodes.test.RunnerTest
+import com.testerum.runner.junit5.logger.JUnitEventLogger
+import com.testerum_api.testerum_steps_api.test_context.ExecutionStatus
+import org.junit.jupiter.api.DynamicContainer.dynamicContainer
+import org.junit.jupiter.api.DynamicNode
+import org.junit.jupiter.api.DynamicTest.dynamicTest
+import org.junit.jupiter.api.fail
+import java.nio.file.Path
+import java.time.LocalDateTime
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.LinkedBlockingDeque
+import kotlin.concurrent.thread
+
+class JunitTestProvider(repositoryDirectory: Path,
```

Comment:
Add comment stating that we expect the tests in JUnit to be executed in the same order as in Testerum, and if this is not the case, this class will do the wrong thing without warning.